reconcile/utils/terrascript_client.py

```diff
@@ -3892,12 +3917,72 @@ def populate_tf_resource_asg(self, resource, namespace_info):
                 {
                     "resource_type": "volume",
                     "tags": tags
-                }
-            ]
+                }]
         }
 
-        region = common_values.get('region') or \
-            self.default_regions.get(account)
+        image = common_values.get('image')
+        image_id = image.get('id')
+        if not image_id:
+            accounts = [a for a in self.filtered_accounts
+                        if a['name'] == account]
+            awsapi = AWSApi(1, accounts,
+                            settings=self.settings,
+                            init_users=False)
+            owner = image.get('owner')
+            if owner in self.uids.keys():
+                owner = self.uids.get(owner)
+
+            url = image.get('url')
+            ref = image.get('ref')
```

Comment:
What happens if there is no `url` or no `ref` in the image? Either handle the case for these calls to `get` returning `None` or, if cannot happen, just do normal dictionary accesses.

Reply:
they are required by schema https://github.com/app-sre/qontract-schemas/pull/53/files#diff-74d10b90e8f97c2e79cc05df43a72a2c59295a88871d5a1f6bbb7af4be7f98d4